src/hotspot/share/memory/heapShared.cpp

```diff
@@ -1106,6 +1107,56 @@ void HeapShared::initialize_subgraph_entry_fields(Thread* THREAD) {
   initialize_preservable_static_field_infos(THREAD);
 }
 
+void HeapShared::initialize_preservable_klass_from_list(Thread* THREAD) {
+  // Initialize preservable classes from given class list, the content format
+  // of class list looks as well as DumpLoadedClassList
+  if (!PreInitializeArchivedClassList) {
+    return;
+  }
+  if (!preinit_classlist_file->is_open()) {
+    log_error(preinit)("Can not open extended preservable class list file %s", PreInitializeArchivedClassList);
+    return;
+  }
+  while (!preinit_classlist_file->eof()) {
+    // Max number of bytes allowed per line in the classlist.
+    // Theoretically Java class names could be 65535 bytes in length. Also, an input line
+    // could have a very long path name up to JVM_MAXPATHLEN bytes in length. In reality,
+    // 4K bytes is more than enough.
+    ResourceMark rm;
+    char buf[4096];
+    char* klass_line = preinit_classlist_file->readln(buf, 4096);
+    if (klass_line == NULL) {
+      continue;
+    }
+    if (strlen(buf) > 0) {
+      Symbol* klass_name = SymbolTable::new_symbol(klass_line, THREAD);
+      Klass* k = SystemDictionary::resolve_or_null(klass_name,
+                                   Handle(THREAD, SystemDictionary::java_system_loader()),
+                                   Handle()/*null_protection_domain*/, THREAD);
+      if (k == NULL) {
+        if (HAS_PENDING_EXCEPTION) {
+          CLEAR_PENDING_EXCEPTION;
+#ifndef PRODUCT
```

Comment:
The exception does not seem to be useful here. Should it be removed?

Reply:
should we change to the following pattern?
```
        if (HAS_PENDING_EXCEPTION) {
#ifndef PRODUCT
          if (Verbose) {
            Handle throwable(THREAD, PENDING_EXCEPTION);
            java_lang_Throwable::print_stack_trace(throwable, tty);
            tty->cr();
          }
#endif
          CLEAR_PENDING_EXCEPTION;
        }
```
to tell details about why resolve_or_null can not load a class.

Reply:
The exception would not be able to give additional useful details on why class loading fails in this case. We normally CLEAR_PENDING_EXCEPTION and print a warning for the class if we cannot load a class at dump time. For example, please see https://github.com/adoptium/jdk11u-fast-startup-incubator/blob/47af1cdd6a2c00f89e93dd50b7ca8f8a9e937165/src/hotspot/share/memory/metaspaceShared.cpp#L1896.

Reply:
Okay, removed.